torchaudio/models/wav2vec2/model.py

```diff
@@ -100,6 +84,44 @@ def forward(
         return x, lengths
 
 
+class Wav2Vec2Model(_BaseModel):
+    """Encoder model used in *wav2vec 2.0* [:footcite:`baevski2020wav2vec`].
+
+    Note:
+        To build the model, please use one of the factory functions.
+
+    Args:
+        feature_extractor (torch.nn.Module):
+            Feature extractor that extracts feature vectors from raw audio Tensor.
+
+        encoder (torch.nn.Module):
+            Encoder that converts the audio features into the sequence of probability
+            distribution (in negative log-likelihood) over labels.
+
+        aux (torch.nn.Module or None, optional):
+            Auxiliary module. If provided, the output from encoder is passed to this module.
+    """
+
+
+class HubertModel(_BaseModel):
```

Comment:
@nateanl 

I started to doubt the value of adding a separate `HubertModel` class. It adds maintenance cost, but practically, it is same as `Wav2Vec2Model`. With the factory functions bellow and pre-trained weights later, I think the existing `Wav2Vec2Model` suffices.

Reply:
Good point. I personally agree that the factory functions are good for now since
1) I think people would try the default configurations of HuBERT (Base, Large, etc) instead of building their own version. Once there is evidence that there is need for hyper-tuning the model, we can provide a more flexible solution then.
2) HuBERT is difficult to train in terms of time, especially for XLarge model. Based on it, the pretrained API is more likely to be used.

Reply:
Okay, I removed the `HubertModel` class. The fact that I did not need to update the unittests I wrote is also a strong evidence that the class was not necessary in the first place.